source/tasks/AwaitTask/AwaitTaskV6/waiter.ts

```diff
@@ -37,26 +44,60 @@ export class Waiter {
         await waiter.waitForServerTasksToComplete(taskIds, inputParameters.pollingInterval * 1000, inputParameters.timeout * 1000, (t) => {
             let context = "";
             const deploymentResult = lookup.get(t.Id);
-            if (deploymentResult?.environmentName) {
-                context = ` to environment '${deploymentResult.environmentName}'`;
-            }
-            if (deploymentResult?.tenantName) {
-                context += ` for tenant '${deploymentResult?.tenantName}'`;
-            }
+            if (deploymentResult) {
+                if (deploymentResult?.environmentName) {
+                    context = ` to environment '${deploymentResult.environmentName}'`;
+                }
+                if (deploymentResult?.tenantName) {
+                    context += ` for tenant '${deploymentResult?.tenantName}'`;
+                }
 
-            if (t.IsCompleted) {
-                this.logger.info?.(`Deployment${context} ${t.State === TaskState.Success ? "completed successfully" : "did not complete successfully"}`);
-            } else {
-                this.logger.info?.(`Deployment${context} is '${t.State}'`);
+                if (t.IsCompleted) {
+                    this.logger.info?.(`Deployment${context} ${t.State === TaskState.Success ? "completed successfully" : "did not complete successfully"}`);
```

Comment:
These are tasks. Could be a deployment, but could also be a Runbook Run. I think the links below will still work the same, but the text here will be misleading.

Reply:
Deployment => Execution?

Reply:
yeah, we've been using that as the generic term to mean either of those things 👍

Reply:
Polling messages updated, task type (deployment or runbook run) is provided by the task:
`Deployment to environment 'dev 2' is 'Executing'`
`Deployment to environment 'test' for tenant 'tenant-EB7CC' is 'Executing'`
`Runbook run to environment 'dev' is 'Executing'`